### PR TITLE
Refactoring Codeblock component [Issue#2158]

### PR DESCRIPTION
**Pull request by esteticalVE**
## Description

Added the possibility to write simple code as plain text  inside Codeblock component

 For example:
image
image

## Related Issue
https://github.com/ethereum/ethereum-org-website/issues/2158

### CODE REVIEW

**Comment by samajammin** [comment]
Hey @esteticalVE - thanks for looking into this? Is this PR still WIP or is this ready for review?

**Comment by esteticalVE** [comment]
Hey @samajammin, ready for review, but I'm not sure about Proptypes, only one icon component uses this

File: src/components/Codeblock.js [hunk 1]
```diff
@@ -1,4 +1,5 @@
 import React, { useState, useContext } from "react"
+import PropTypes from "prop-types"
```
Comment (wackerow) [inline src/components/Codeblock.js:2]: Maybe I'm missing something, but I'm not sure this is required here. Testing locally without it seems to work just fine in the code blocks I checked.  Was there a reason for this?

File: src/components/Codeblock.js [hunk 2]
```diff
@@ -298,4 +322,8 @@ const Codeblock = (props) => {
   )
 }
 
+Codeblock.propTypes = {
+  children: PropTypes.node.isRequired,
+}
+
```
Comment (wackerow) [inline src/components/Codeblock.js:328]: (see above)

File: src/components/Codeblock.js [hunk 3]
```diff
@@ -214,26 +215,49 @@ const codeTheme = {
...
+  }).join("")
+
   const [isCollapsed, setIsCollapsed] = useState(true)
-  const className = props.children.props.className || ""
+  const className =
+    props.children?.props?.className || props?.codeLanguage || ""
   const matches = className.match(/language-(?<lang>.*)/)
-  const language =
-    matches && matches.groups && matches.groups.lang ? matches.groups.lang : ""
+  const language = matches?.groups?.lang || ""
```
Comment (wackerow) [inline src/components/Codeblock.js:246]: Thank you for cleaning this up =)

**Review by wackerow — COMMENTED** [review]
@esteticalVE Hey, this looks great! Working great locally. (Just made one note about using `PropTypes`... this needed?)

Note to all: To avoid prettier/auto-formatting screwing up the code when you save, it's often best to wrap the code in {``}, and don't indent new lines, ie:
```
<Codeblock>
  {`const hello = "Hello"
const world = "World"`}
</Codeblock>
```
👍

File: src/components/Codeblock.js (hunk 1 above)
Comment (esteticalVE) [reply src/components/Codeblock.js:2]: This is for developers, so as not to leave the Codeblock empty,  but i can remove it

**Comment by esteticalVE** [comment]
@wackerow Hey! This is for developers, so as not to leave the Codeblock empty, but i can remove it

**Comment by samajammin** [comment]
Thanks again for this @esteticalVE! Sorry for the delay here, reviewing now.

**Review by samajammin — APPROVED** [review]
Looks great! Thanks for knocking this out 🙏

**Comment by samajammin** [comment]
@all-contributors please add @esteticalVE for code.

**Pull request merged**